pydantic_ai_slim/pydantic_ai/_parts_manager.py

```diff
@@ -67,20 +118,49 @@ def get_parts(self) -> list[ModelResponsePart]:
         """
         return [p for p in self._parts if not isinstance(p, ToolCallPartDelta)]
 
-    def handle_text_delta(
+    def handle_text_delta(  # noqa: C901
         self,
         *,
         vendor_part_id: VendorId | None,
         content: str,
         id: str | None = None,
-        thinking_tags: tuple[str, str] | None = None,
+        thinking_tags: ThinkingTags | None = None,
         ignore_leading_whitespace: bool = False,
-    ) -> ModelResponseStreamEvent | None:
+    ) -> Sequence[ModelResponseStreamEvent]:
         """Handle incoming text content, creating or updating a TextPart in the manager as appropriate.
 
-        When `vendor_part_id` is None, the latest part is updated if it exists and is a TextPart;
-        otherwise, a new TextPart is created. When a non-None ID is specified, the TextPart corresponding
-        to that vendor ID is either created or updated.
+        This function also handles what we'll call "loose thinking", which is the generation of
+        ThinkingParts via explicit thinking tags embedded in the text content.
+        Activating loose thinking requires:
+        - `thinking_tags` to be provided, which is a tuple of (opening_tag, closing_tag)
```

Comment:
Please wrap all code in backticks, i.e. `(opening_tag, closing_tag)` here, `ThinkingPart` above, `vendor_part_id` below, etc